crates/node-db/tests/query.rs

```diff
@@ -122,6 +122,32 @@ fn test_list_blocks_by_time() {
     assert_eq!(expected_blocks, fetched_blocks);
 }
 
+#[test]
+fn get_block_header() {
+    let blocks = util::test_blocks(10);
+    let headers: Vec<_> = blocks.iter().map(|b| (b.number, b.timestamp)).collect();
+
+    // Create an in-memory SQLite database.
+    let mut conn = test_conn();
+
+    // Create the necessary tables and insert blocks.
+    let tx = conn.transaction().unwrap();
+    node_db::create_tables(&tx).unwrap();
+    for block in &blocks {
+        node_db::insert_block(&tx, block).unwrap();
+    }
+    tx.commit().unwrap();
+
+    // Fetch the headers and check they match.
+    let fetched_headers: Vec<_> = blocks
+        .iter()
+        .map(essential_hash::content_addr)
+        .map(|ca| node_db::get_block_header(&conn, &ca).unwrap().unwrap())
+        .collect();
+
+    assert_eq!(&headers, &fetched_headers);
```

Comment:
Maybe we should replace instances in which we check the equality of `Vec<Block>`s or other essential types that implement `Eq` in some tests